docs/language/functions.rst

```diff
@@ -353,6 +353,32 @@ values. Here is an example of an overloaded function:
 In the function foo, abs() is called with an ``int64`` so the second implementation
 of the function abs() is called.
 
+.. note::
+  The substrate target runtime requires function names to be unique.
+  Overloaded function names will be mangled in the ABI. 
+  The function name will be concatenated with all of its argument types, seperated by underscores.
+  Struct types are represented by their field types (preceded with an extra underscore at the start).
+  Enum types are represented as their underlying uint8 type.
+
+  The following example illustrates some overloaded functions and their mangled name:
+
+  .. code-block:: solidity
+
+    enum E { v1, v2 }
+    struct S { int256 i; bool b; address a; }
+
+    contract C {
+        // foo_
+        function foo() public pure {}
+
+        // foo_uint256_address[]
```

Comment:
I don't think `[]` wil be allowed in metadata message names. What will the function name in typescript/rust?

Reply:
Are you sure? In polkadot.js/apps and the contracts UI, this works. The names are represented as `fooUint256Address[]` resp. `fooUint8[2]Int256BoolAddress`, so this seems fine.

Reply:
I thought a bit about this, but couldn't find a satisfying solution. We could just remove the square brackets, but then dynamic arrays would look the same as non-array types. I'm not sure what a good substitute would be either.

Reply:
It may work in the contracts UI, but what about the using a contract in typescript using `@polkadot/api-contract`? What will the function be called, i.e. `contract.query.fooUint256Address[]` is not valid in typescript.

How about replacing `[` with `_a` and `]` with ``.